pkg/providers/amifamily/ami.go

```diff
@@ -132,6 +132,38 @@ func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTempla
 	return amiIDs, nil
 }
 
+func (p *Provider) GetAMIWithRequirements(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, amiFamily AMIFamily) (map[AMI]scheduling.Requirements, error) {
+	kubernetesVersion, err := p.KubeServerVersion(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting kubernetes version %w", err)
+	}
+	amiIDs := map[string][]*cloudprovider.InstanceType{}
+	amiRequirements, err := p.getAMIRequirements(ctx, nodeTemplate)
+	if err != nil {
+		return nil, err
+	}
+	if len(amiRequirements) == 0 {
+		for _, instanceType := range instanceTypes {
```

Comment:
This seems unneccesary. We know that there are at-most three AMIs that will be produced from this call so there's no need to iterate through the instance types.